authentication/api_gateway/serializers.py

```diff
@@ -33,9 +34,14 @@ def create(self, validated_data):
         legal_address_data = validated_data.pop("legal_address")
         user_profile_data = validated_data.pop("user_profile", None)
         with transaction.atomic():
-            user.openedx_user.edx_username = username
             user.name = name
             user.save()
+            if user.openedx_user is None:
+                create_user(user)
```

Comment:
When I tested this with a brand-new user, I got an error back from edX because there wasn't a username specified. I had supplied one in `/create-profile/` in mitxonline but there wasn't yet an `OpenEdxUser` record for my user, so no username got passed on to edX. So, I think we should explicitly create the `OpenEdxUser` if there's not one already. (`create_user` will do `get_or_create` but since the username's not there in a new record, it fails.)